### PR TITLE
os/mm : Merge duplicated code about remove node

### PR DESCRIPTION
**Pull request by YiHyunJin**
There are many duplicated code.
It can be merged into defines for avoiding redundancy.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [bf9f37e7624180e6b581d70f0676b887db01b0a2] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [bf9f37e7624180e6b581d70f0676b887db01b0a2] - Code Rule Check (C++) OK.

File: os/include/tinyara/mm/mm.h [hunk 1]
```diff
@@ -245,6 +245,15 @@ typedef size_t mmsize_t;
 #define MMSIZE_MAX SIZE_MAX
 #endif
 
+#define REMOVE_NODE(a)					\
```
Comment (sunghan-chang) [inline os/include/tinyara/mm/mm.h]: 1. Let's remove it to say what it does.
   `REMOVE_NODE_FROM_LIST` or `REMOVE_NODE_IN_LIST` or something else
2. It is not called from outside. Let's add this in the header of os/mm/mm_heap folder.

**Comment by seinfra** [comment]
Target : [c87224638b82ce97403a326b7e06fae6c694b1b8] - Code Rule Check (C++) OK.

File: os/mm/mm_heap/mm_heap.h [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+#ifndef __MM_MM_HEAP_MM_HEAP_H
+#define __MM_MM_HEAP_MM_HEAP_H
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+
+#include <tinyara/mm/mm.h>
```
Comment (sunghan-chang) [inline os/mm/mm_heap/mm_heap.h]: Both are unnecessary. 
<debug.h> is necessary.

Comment (sunghan-chang) [reply os/mm/mm_heap/mm_heap.h]: Let's name it using what it does. mm_queue.h ? mm_node.h?

**Comment by seinfra** [comment]
Target : [3257f332f0e4d79194f9d2e3254b259e5ee33249] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [3257f332f0e4d79194f9d2e3254b259e5ee33249] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [ee4b5d77f54b60eafcefc29834bc59b656738b3b] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [ee4b5d77f54b60eafcefc29834bc59b656738b3b] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [a74156f4f7ff3fa9525a8c1b61ea543ead9842cf] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [a74156f4f7ff3fa9525a8c1b61ea543ead9842cf] - Code Rule Check OK.

File: os/mm/mm_heap/mm_node.h [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+ * Included Files
+ ****************************************************************************/
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#define REMOVE_NODE_FROM_LIST(node)				\
+	do {							\
+		DEBUGASSERT((node)->blink);			\
```
Comment (sunghan-chang) [inline os/mm/mm_heap/mm_node.h]: include <debug.h>

**Comment by seinfra** [comment]
Target : [fdd1b724f9c96861181fce8ace219a3e60d97ea2] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [fdd1b724f9c96861181fce8ace219a3e60d97ea2] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [835b24a06363640d1fa3d862521c628b134a0074] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [835b24a06363640d1fa3d862521c628b134a0074] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [6c96d7940f2a773c27c410832708351a7d01dddc] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [6c96d7940f2a773c27c410832708351a7d01dddc] - Code Rule Check (C++) OK.

File: os/mm/mm_heap/mm_node.h (hunk 3 above)
Comment (YiHyunJin) [reply os/mm/mm_heap/mm_node.h]: I think it should add <assert.h>.

**Pull request merged**